buildkite/provider.go

```diff
@@ -16,6 +25,108 @@ const (
 	SchemaKeyRestURL      = "rest_url"
 )
 
+type terraformProvider struct {
+	version string
+}
+
+type providerModel struct {
+	ApiToken     types.String `tfsdk:"api_token"`
+	GraphqlUrl   types.String `tfsdk:"graphql_url"`
+	Organization types.String `tfsdk:"organization"`
+	RestUrl      types.String `tfsdk:"rest_url"`
+}
+
+func (tf *terraformProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+	var data providerModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	apiToken := os.Getenv("BUILDKITE_API_TOKEN")
+	graphqlUrl := defaultGraphqlEndpoint
+	organization := os.Getenv("BUILDKITE_ORGANIZATION")
+	restUrl := defaultRestEndpoint
+
+	if data.ApiToken.ValueString() != "" {
+		apiToken = data.ApiToken.ValueString()
+	}
+	if data.GraphqlUrl.ValueString() != "" {
+		graphqlUrl = data.GraphqlUrl.ValueString()
+	} else if v, ok := os.LookupEnv("BUILDKITE_GRAPHQL_URL"); ok {
+		graphqlUrl = v
+	}
+	if data.Organization.ValueString() != "" {
+		organization = data.Organization.ValueString()
+	}
+	if data.RestUrl.ValueString() != "" {
+		restUrl = data.RestUrl.ValueString()
+	} else if v, ok := os.LookupEnv("BUILDKITE_REST_URL"); ok {
+		restUrl = v
+	}
+
+	legacyProvider := schema.Provider{}
+	config := clientConfig{
+		apiToken:   apiToken,
+		graphqlURL: graphqlUrl,
+		org:        organization,
+		restURL:    restUrl,
+		userAgent:  legacyProvider.UserAgent("buildkite", tf.version),
+	}
+	client, err := NewClient(&config)
+
+	if err != nil {
+		resp.Diagnostics.AddError(err.Error(), fmt.Sprintf("... details ... %s", err))
+	}
+
+	resp.ResourceData = &client
+	resp.DataSourceData = &client
+}
+
+func (*terraformProvider) DataSources(context.Context) []func() datasource.DataSource {
+	return []func() datasource.DataSource{}
+}
+
+func (tf *terraformProvider) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = "buildkite"
+	resp.Version = tf.version
+}
+
+func (*terraformProvider) Resources(context.Context) []func() resource.Resource {
+	return []func() resource.Resource{}
+}
+
+func (*terraformProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+	resp.Schema = framework_schema.Schema{
+		Attributes: map[string]framework_schema.Attribute{
+			SchemaKeyOrganization: framework_schema.StringAttribute{
+				Optional:    true,
+				Description: "The Buildkite organization slug",
+				// MarkdownDescription: "The Buildkite organization slug",
```

Comment:
Are we keeping these in for any reason? The `Description` maintains the same value and isn't unclear in what it is.

Reply:
The `MarkdownDescription` would be in the commit history too, so I don't see a reason to just keep old code around.

Reply:
I agree with @mcncl on this one.

Reply:
Oh yeah we can add these in later really. The problem is with this muxing between sdkv2 and framework, the provider definition has to match _exactly_!
So I wanted to include `MarkdownDescription` because apparently we can start to automatically generate documentation which I think will be better. But sdkv2 doesn't have this property and it was breaking with it included.
Its the same reason I had to remove `DefaultFunc` from the sdkv2 provider, because that doesn't exist in framework now 🤷‍♂️

Once we have fully migrated to framework we could uncomment these. But I'll just create a task to come back to docs generation after the migration. And then we can remove them now